pkg/client/cli/cliutil/systema.go

```diff
@@ -3,33 +3,57 @@ package cliutil
 import (
 	"context"
 	"errors"
-	"os"
+
+	grpcCodes "google.golang.org/grpc/codes"
+	grpcStatus "google.golang.org/grpc/status"
+	empty "google.golang.org/protobuf/types/known/emptypb"
 
 	"github.com/telepresenceio/telepresence/rpc/v2/connector"
-	"github.com/telepresenceio/telepresence/v2/pkg/client/auth"
 	"github.com/telepresenceio/telepresence/v2/pkg/client/auth/authdata"
 )
 
 // EnsureLoggedIn ensures that the user is logged in to Ambassador Cloud.  An error is returned if
 // login fails.  The result code will indicate if this is a new login or if it resued an existing
 // login.
 func EnsureLoggedIn(ctx context.Context) (connector.LoginResult_Code, error) {
-	return auth.EnsureLoggedIn(ctx, os.Stdout)
+	var resp *connector.LoginResult
+	err := WithConnector(ctx, func(ctx context.Context, connectorClient connector.ConnectorClient) error {
+		var err error
+		resp, err = connectorClient.Login(ctx, &empty.Empty{})
+		return err
+	})
+	return resp.GetCode(), err
 }
 
-// Logout logs out of Ambassador Cloud.  Returns auth.ErrNotLoggedIn if not logged in.
+// Logout logs out of Ambassador Cloud.  Returns an error if not logged in.
 func Logout(ctx context.Context) error {
-	return auth.Logout(ctx)
+	err := WithConnector(ctx, func(ctx context.Context, connectorClient connector.ConnectorClient) error {
+		_, err := connectorClient.Logout(ctx, &empty.Empty{})
+		return err
+	})
+	if grpcStatus.Code(err) == grpcCodes.NotFound {
+		err = errors.New(grpcStatus.Convert(err).Message())
+	}
+	if err != nil {
```

Comment:
it would seem we are effectively returning `err` in either case bc we return `err` if err != nil and `nil` if err == nil.  Is this done to make it more legible and clear that we are explicitly returning an err? Or is it something we can optimize?

Reply:
Yes, it's done for style reasons.  Part of that is legibility.  Part of that is that there are two cases, so having it this way lets coverage testing tell us if we're testing both cases (not that we're doing coverage testing on telepresence right now).  Part of it is paranoia about "nil isn't always nil".